draw.it.client/src/pages/host/HostScreen.jsx

```diff
@@ -124,17 +126,42 @@ function HostScreen() {
         debouncedSend(newCatId, drawingTime, numberOfRounds, roomName, addAiPlayer);
     };
 
+    const RULES = Object.freeze({
+        drawingTime: { min: 20, max: 180, step: 1 },
+        numberOfRounds: { min: 1, max: 10, step: 1 },
+    });
+
+    const clampAndSnap = (val, { min, max, step }) => {
+        let v = Number(val);
+        if (Number.isNaN(v)) v = min;
+        v = Math.min(max, Math.max(min, v));
+        const base = min ?? 0;
+        if (step && step > 0) {
+            v = Math.round((v - base) / step) * step + base;
+        }
+        return v;
+    };
+
+    // Waits 500ms after the last input before clamping and snapping the value
+    const debouncedClampAndSnap = useMemo(() => {
+        return debounce((val, rules, setter, fieldName) => {
+            const newValue = clampAndSnap(val, rules);
+            setter(newValue);
+
+            if (fieldName === 'drawingTime') {
+                debouncedSend(categoryId, newValue, numberOfRounds, roomName, addAiPlayer);
+            } else if (fieldName === 'numberOfRounds') {
+                debouncedSend(categoryId, drawingTime, newValue, roomName, addAiPlayer);
+            }
+        }, 1000);
+    }, [categoryId, drawingTime, numberOfRounds, roomName, addAiPlayer, debouncedSend]);
```

Comment:
I don't know why you decided to change this, but please make it `[lobbyConnection, roomId]` again
```suggestion
    }, [lobbyConnection, roomId]);
```